### PR TITLE
Fix/knn filter

### PR DESCRIPTION
**Pull request by djklim87**

### CODE REVIEW

File: src/Plugin/Knn/Handler.php [hunk 1]
```diff
@@ -178,6 +156,33 @@ private static function knnSqlQuery(Client $manticoreClient, Payload $payload, s
 			}
 		}
 
+
 		return $result;
 	}
+
+	/**
```
Comment (donhardman) [inline src/Plugin/Knn/Handler.php:163]: Missing doc info about this method cuz from the name it's actually not very clear what it does for me when you are no in the context

File: src/Plugin/Knn/Handler.php [hunk 2]
```diff
@@ -178,6 +156,33 @@ private static function knnSqlQuery(Client $manticoreClient, Payload $payload, s
...
+	 * @param Payload $payload
+	 * @param string $queryVector
+	 * @return void
+	 */
+	private static function substituteParsedQuery(Payload $payload, string $queryVector): void {
+
+		$parsedQuery = $payload::$sqlQueryParser::getParsedPayload();
+		if (!is_array($parsedQuery)) {
```
Comment (donhardman) [inline src/Plugin/Knn/Handler.php:171]: In what case it's not array? Failed to parse?

File: src/Plugin/Knn/Payload.php [hunk 3]
```diff
@@ -29,51 +29,84 @@ final class Payload extends BasePayload
 	/** @var array<string> */
 	public array $select = [];
 
+	/** @var string|null table */
```
Comment (donhardman) [inline src/Plugin/Knn/Payload.php:32]: Prefer `?string $table`

File: src/Plugin/Knn/Payload.php [hunk 4]
```diff
@@ -29,51 +29,84 @@ final class Payload extends BasePayload
...
 		return $self;
 	}
 
 	/**
-	 * @param  Request  $request
+	 * @param Payload $self
+	 * @param Request $request
+	 * @return void
+	 */
+	private static function parseHttpRequest(self $self, Request $request): void {
```
Comment (donhardman) [inline src/Plugin/Knn/Payload.php:63]: `$self` is not very nameable for here IMHO, prefer to use `$payload`

File: src/Plugin/Knn/Payload.php [hunk 5]
```diff
@@ -29,51 +29,84 @@ final class Payload extends BasePayload
...
+		$self->field = $payload['knn']['field'];
+		$self->k = (string)$payload['knn']['k'];
+		$self->docId = (string)$payload['knn']['doc_id'];
+	}
+
+	/**
+	 * @param Payload $self
+	 * @return void
+	 */
+	private static function parseSqlRequest(self $self): void {
```
Comment (donhardman) [inline src/Plugin/Knn/Payload.php:87]: Prefer `$payload` instead of `$self`

**Review by donhardman — CHANGES_REQUESTED** [review]
Sounds good, just some small changes and missing doc to be fixed

File: src/Plugin/Knn/Handler.php (hunk 2 above)
Comment (djklim87) [reply src/Plugin/Knn/Handler.php:171]: Yes, I changed this condition to comparison with null

File: src/Plugin/Knn/Payload.php (hunk 4 above)
Comment (djklim87) [reply src/Plugin/Knn/Payload.php:63]: Bottom we already use this word for `$request->payload`... So in the current method, it has reason to use self

File: src/Plugin/Knn/Payload.php (hunk 5 above)
Comment (djklim87) [reply src/Plugin/Knn/Payload.php:87]: the same as before

File: src/Plugin/Knn/Payload.php (hunk 3 above)
Comment (djklim87) [reply src/Plugin/Knn/Payload.php]: done

File: src/Plugin/Knn/Payload.php (hunk 4 above)
Comment (djklim87) [reply src/Plugin/Knn/Payload.php:71]: I changed naming, but IMHO it make understanding of code more complex

**Pull request merged**